### PR TITLE
add initial working m5stack_stamps3

### PR DESCRIPTION
**Pull request by jamesjnadeau**
## Checklist

*By completing this PR sufficiently, you help us to review this Pull Request quicker and also help improve the quality of Release Notes*

- [X] Please provide specific title of the PR describing the change
- [X] Please provide related links (*eg. Issue which will be closed by this Pull Request*)
- [X] If you are adding an new boards, please make sure
  - [X] [Provide link to your allocated VID/PID if applicable](https://github.com/espressif/usb-pids/blob/main/allocated-pids.txt#L290)
  - [X] Add your board to [action ci](/.github/workflows) in correct workflow and alphabet order for release binary
  - [X] `UF2_BOARD_ID` in your board.h follow correct format from [uf2 specs](https://github.com/microsoft/uf2#files-exposed-by-bootloaders)

*This checklist items that are not applicable to your PR can be deleted.*

-----------

## Description of Change

Adding setup for m5 stack stamp s3

### CODE REVIEW

File: ports/espressif/boards/m5stack_stamps3/board.h [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+// #ifndef ADAFRUIT_FEATHER_ESP32S3_TFT_H_
```
Comment (hathach) [inline ports/espressif/boards/m5stack_stamps3/board.h:25]: please remove those unused, and add header defie

File: ports/espressif/boards/m5stack_stamps3/board.h [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+#define USB_MANUFACTURER         "M5"
+#define USB_PRODUCT              "Stamp S3"
+
+#define UF2_PRODUCT_NAME         USB_MANUFACTURER " " USB_PRODUCT
+#define UF2_BOARD_ID             "ESP32S3"
+#define UF2_VOLUME_LABEL         "M5S3BOOT"
+#define UF2_INDEX_URL            "https://www.adafruit.com/product/5483"
+
+// Use favicon
+#define TINYUF2_FAVICON_HEADER   "favicon_adafruit_256.h"
```
Comment (hathach) [inline ports/espressif/boards/m5stack_stamps3/board.h:116]: please remove the use of adafruit logo for non-adafruit board

File: ports/espressif/boards/m5stack_stamps3/board.h [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+//--------------------------------------------------------------------+
+
+#define USB_VID                  0x303A
+#define USB_PID                  0x811A
+
+#define USB_MANUFACTURER         "M5"
+#define USB_PRODUCT              "Stamp S3"
+
+#define UF2_PRODUCT_NAME         USB_MANUFACTURER " " USB_PRODUCT
+#define UF2_BOARD_ID             "ESP32S3"
```
Comment (hathach) [inline ports/espressif/boards/m5stack_stamps3/board.h:111]: incorrect board-id format

File: ports/espressif/boards/m5stack_stamps3/board.h [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+#define UF2_VOLUME_LABEL         "M5S3BOOT"
+#define UF2_INDEX_URL            "https://www.adafruit.com/product/5483"
```
Comment (hathach) [inline ports/espressif/boards/m5stack_stamps3/board.h:113]: incorrect product url

**Review by hathach — CHANGES_REQUESTED** [review]

**Comment by jamesjnadeau** [comment]
thank you @hathach, I think this is all set now, please let me know if I missed anything.

File: ports/espressif/boards/m5stack_stamps3/board.h (hunk 3 above)
Comment (jamesjnadeau) [reply ports/espressif/boards/m5stack_stamps3/board.h:94]: I have updated the board folder from cardputer to stamps3 to better match the board.

File: ports/espressif/boards/m5stack_stamps3/board.h (hunk 1 above)
Comment (hathach) [reply ports/espressif/boards/m5stack_stamps3/board.h]: header define still not added

File: ports/espressif/boards/m5stack_stamps3/board.h (hunk 3 above)
Comment (hathach) [reply ports/espressif/boards/m5stack_stamps3/board.h:94]: board-id still not in correct format

**Review by hathach — CHANGES_REQUESTED** [review]
2 of change requests are still not fixed

File: ports/espressif/boards/m5stack_stamps3/board.h (hunk 3 above)
Comment (panguin6010) [reply ports/espressif/boards/m5stack_stamps3/board.h:94]: ```#define UF2_BOARD_ID             "ESP32S3-StampS3-01"```

Here is the correct board id, i didnt want to make another PR

File: ports/espressif/boards/m5stack_stamps3/board.h (hunk 1 above)
Comment (panguin6010) [reply ports/espressif/boards/m5stack_stamps3/board.h]:
```
#ifndef M5STACK_STAMPS3_BOARD_H_
#define M5STACK_STAMPS3_BOARD_H_``
```
put those under the license

`#endif `
put this at the end

**Comment by panguin6010** [comment]
@jamesjnadeau In leu of making another PR I commented the fixes for the open changes

**Comment by jamesjnadeau** [comment]
@hathach this should now be fixed, thanks @panguin6010  for the help!

**Review by hathach — APPROVED** [review]
look good

**Pull request merged**